api/samlacs.go

```diff
@@ -148,7 +154,24 @@ func (a *API) SAMLACS(w http.ResponseWriter, r *http.Request) error {
 		}
 	}
 
-	// TODO: fetch new metadata if possible when validUntil < time.Now() or cacheDuration
+	var samlMetadataModified bool
+	samlMetadataModified = false
+	if IsMetadataStale(idpMetadata, ssoProvider.SAMLProvider) {
+		if *ssoProvider.SAMLProvider.MetadataURL != "" {
+			rawMetadata, err := fetchSAMLMetadata(ctx, *ssoProvider.SAMLProvider.MetadataURL)
+			if err != nil {
+				// Fail silently but raise warning and continue with existing metadata
+				logentry := log.WithField("sso_provider_id", ssoProvider.ID.String())
+				logentry = logentry.WithField("saml_provider_metadata", ssoProvider.SAMLProvider.MetadataXML)
```

Comment:
Probably not the best idea to log the whole metadata here. It could be huge. What about just logging the timestamps?

Reply:
Makes sense, would this be the CreatedAt and UpdatedAt timestamps?
```suggestion
				logentry = logentry.WithField("saml_provider_metadata_creation_time", ssoProvider.SAMLProvider.CreatedAt)
				logentry = logentry.WithField("saml_provider_metadata_updated_time", ssoProvider.SAMLProvider.UpdatedAt)
```